CMakeLists.txt

```diff
@@ -87,6 +87,9 @@ endif ()
 project(gdal LANGUAGES C CXX)
 include(CTest)
 
+# Include defaults
+include ("${CMAKE_SOURCE_DIR}/cmake/ConfigUser.cmake")
```

Comment:
Is it a common / best-practice to do that ? Are other projects doing such thing ?
And the use of CMAKE_SOURCE_DIR is incorrect here as it breaks adding GDAL as a sub project. This should rather be CMAKE_CURRENT_SOURCE_DIR or PROJECT_SOURCE_DIR.

Reply:
actually I see that this need should be covered by the -C switch of CMake:
```
       -C <initial-cache>
              Pre-load a script to populate the cache.

              When  CMake  is  first  run in an empty build tree, it creates a CMakeCache.txt file and populates it with customizable settings for the project.  This option may be used to specify a
              file from which to load cache entries before the first pass through the project’s CMake listfiles.  The loaded entries take priority over the project’s default values.  The given file
              should be a CMake script containing set() commands that use the CACHE option, not a cache-format file.
```
Using that rather cmake/ConfigUser.cmake would avoid having in version control a file that users could modify and thus accidentally commit if they are GDAL developers.

The https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html newer functionality also seems close to that

Reply:
GMT and NetCDF use it, though NetCDF picked it from GMT and put the ``ConfigUser.cmake`` under version control, whilst GMT says is up to the user to create that file (and it has cmake logic to create an empty one to not error).

I'm fine with any solution that makes it simpler and also don't like much the idea of putting a file under version control and then tell the users to modify it at their own need.
Perhaps you can improve this bit yourself.